engine/routes.py

```diff
@@ -50,12 +51,13 @@ def handle_exception(error):
 @app.before_request
 def log_request():
     data = request.get_json()
-    request_id = data.get('requestUUID')
-    streams = data.get('streams')
-    if log.isEnabledFor(logging.DEBUG):
+    if data:
```

Comment:
How about this:

```python
data = request.get_json()
request_id = None
streams = None
if data:
  request_id = data.get('requestUUID')
  streams = data.get('streams')
log.debug('<%s> Incoming request url=%r data=%r', request_id, request.url, data)
log.info('<%s> Handling request to %r - %r', request_id, request.url, streams)
```
or
```python
data = request.get_json()
request_id = data.get('requestUUID') if data else None
streams = data.get('streams') if data else None
log.debug('<%s> Incoming request url=%r data=%r', request_id, request.url, data)
log.info('<%s> Handling request to %r - %r', request_id, request.url, streams)
```

I suspect the first is optimal, but I haven't tested it.

Reply:
Tested that the first is faster. Updating...

Reply:
Ooo - I should be a cookie for that...

Reply:
In general, else statements are horrible for code optimization. It forces a branch which is expensive at the machine code level.